Userland/Games/Minesweeper/CustomGameDialog.cpp

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2021, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "CustomGameDialog.h"
+#include "Field.h"
+#include <AK/String.h>
+#include <Games/Minesweeper/MinesweeperCustomGameWindowGML.h>
+#include <LibGUI/BoxLayout.h>
+#include <LibGUI/Button.h>
+#include <LibGUI/ComboBox.h>
+#include <LibGUI/ItemListModel.h>
+#include <LibGUI/Label.h>
+#include <LibGUI/MessageBox.h>
+#include <LibGUI/SpinBox.h>
+#include <LibGUI/Statusbar.h>
+#include <LibGUI/TextBox.h>
+#include <LibGUI/Widget.h>
+
+int CustomGameDialog::show(GUI::Window* parent_window, Field& field)
+{
+    auto dialog = CustomGameDialog::construct(parent_window);
+
+    if (parent_window) {
+        dialog->set_icon(parent_window->icon());
+        dialog->center_within(*parent_window);
+    }
+
+    dialog->m_columns_spinbox->set_value(field.columns());
+    dialog->m_rows_spinbox->set_value(field.rows());
+    dialog->m_mines_spinbox->set_value(field.mine_count());
+
+    auto result = dialog->exec();
+
+    if (result != GUI::Dialog::ExecOK)
+        return result;
+
+    field.set_field_size(dialog->m_rows_spinbox->value(), dialog->m_columns_spinbox->value(), dialog->m_mines_spinbox->value());
+
+    return GUI::Dialog::ExecOK;
+}
+
+void CustomGameDialog::set_max_mines()
+{
+    // Generating a field with > 50% mines takes too long.
+    // FIXME: Allow higher amount of mines to be placed.
+    m_mines_spinbox->set_max((m_rows_spinbox->value() * m_columns_spinbox->value()) / 2);
+}
+
+CustomGameDialog::CustomGameDialog(Window* parent_window)
+    : Dialog(parent_window)
+{
+    resize(300, 90);
```

Comment:
As visible in your gif, (and I tested it myself) this wants to be a little wider, since the mines SpinBox is squeezed against the right edge of the GroupBox.